packages/ai/src/authorizers/federated-connections/FederatedConnectionAuthorizerBase.ts

```diff
@@ -65,11 +69,15 @@ export class FederatedConnectionAuthorizerBase<ToolExecuteArgs extends any[]> {
       );
     }
 
-    // Validate resource server client credentials when using access tokens
-    if (hasAccessToken) {
+    // Validate resource server client credentials when using access tokens for federated connections
+    if (
+      hasAccessToken &&
+      hasSubjectTokenType &&
+      params.subjectTokenType === SUBJECT_TOKEN_TYPES.SUBJECT_TYPE_ACCESS_TOKEN
+    ) {
       if (!this.auth0.clientId || !this.auth0.clientSecret) {
         throw new Error(
-          "clientId and clientSecret must be provided when using accessToken for federated token exchange."
+          "clientId and clientSecret must currently be provided when using accessToken for federated token exchange."
```

Comment:
noting that we can also add support for `clientAssertionSigningKey`, but considering that out of scope for the current iteration.
https://github.com/auth0/auth0-auth-js/pull/39/files#diff-9b017b5bb122fc1f6077bc35bcd33feb05b1a75b86cba83223f2e135fbf9e43a